docs-v2/guides/syncs/detecting-deletes.mdx

```diff
@@ -110,45 +104,48 @@ export default createSync({
     for await (const page of tickets) {
       await nango.batchSave(page, 'Ticket');
     }
+
+    // Delete records that don't exist anymore
+    await nango.deleteRecordsFromPreviousExecution('Ticket');
   }
 });
 ```
 
 ### How the algorithm works
 
-1. After a successful run, Nango stores the list of record IDs.
-2. On the next run, Nango compares the new list with the old one.
+1. During each execution, Nango stores the list of record IDs.
+2. When calling `deleteRecordsFromPreviousExecution`, Nango compares the new list with the old one.
 3. Any records missing in the new list are marked as deleted (soft delete). They remain accessible from the Nango cache, but with `record._metadata.deleted === true`.
 
 <Warning>
-**Be careful with exception handling when using** `trackDelete`
+**Be careful with exception handling when using** `deleteRecordsFromPreviousExecution`
 
 Nango only performs deletion detection (the “diff”) if a sync run completes successfully without any uncaught exceptions.
```

Comment:
Is this still true? 

If they call `deleteRecordsFromPreviousExecution`, then the script crashes, the diff won't happen? From what I read below, I imagine the latter? 

So even the script succeeds, I imagine the diffing operation will not be executed when the method is called, but after script completion? Probably worth making this explicit if it's not already.

Reply:
you are right. This is not entirely true anymore. I will fix